funding/manager.go

```diff
@@ -1039,13 +1056,85 @@ func (f *Manager) stateStep(channel *channeldb.OpenChannel,
 	return fmt.Errorf("undefined channelState: %v", channelState)
 }
 
+// waitForFundingSpend waits for any of the inputs to the provided funding
+// transaction to be spent, and returns the txid of the spending transaction.
+// This is used to detect the case where the funding tx is double spent and
+// rendered invalid.
+func (f *Manager) waitForFundingSpend(fundingTx *wire.MsgTx,
+	fndBroadcastHeight uint32) (*chainhash.Hash, error) {
+
+	quit := make(chan struct{})
+	defer close(quit)
+
+	errs := make(chan error)
+	spends := make(chan *chainntnfs.SpendDetail)
+
+	for _, txIn := range fundingTx.TxIn {
+
+		f.wg.Add(1)
+		go func() {
+			defer f.wg.Done()
+
+			pkScript, err := txscript.ComputePkScript(
```

Comment:
IIRC, the script here is actually optional? Will need to double check.

Reply:
Not sure what you mean by can be optional, I'm not sure how else I'd detect the double spend if this wasn't populated

Reply:
```
spendNtfn, err := f.cfg.Notifier.RegisterSpendNtfn(
				&txIn.PreviousOutPoint, []byte{},
				fndBroadcastHeight,
			)
```

The itest I added fails without the `pkScript`

Reply:
Yes the script has to be valid so your approach is correct

Reply:
I think it's possible to construct a PSBT that funds the channel with a p2tr-spending input. That would mean this logic would break since `ComputePkScript` does not understand p2tr at the moment

Reply:
So, I think this pull should wait for `btcd/txscript`s `ComputePkScript` function to be updated with taproot logic

Reply:
This can now use `RegisterSpendNtfn` with the zero-taproot pkscript for all inputs. This is hacky, but this will match on all pkscripts so it only takes the outpoint into consideration. It should use some buffer (12 blocks) for the height hint (so broadcastheight-12) in case the funding flow was slow and one of the inputs has already been spent. 

Since `RegisterSpendNtfn` only notifies after 1 block, a call should be made to `RegisterConfirmationsNtfn` on the spent txid with a conf number of `numconfs` (whatever the channel has decided). This ensures that lnd has more assurance that the spend isn't reorged out and replaced with the funding txn. Then the function should return to the caller.